references/detection/train_tpu.py

```diff
@@ -128,28 +149,23 @@ def process_prediction(output, threshold=0.05):
     return masks, pred_boxes, pred_class
 
 
-def do_prediction(image_path, use_cpu=False):
-    model = torchvision.models.detection.maskrcnn_resnet50_fpn(pretrained=True)
-    model.eval()
-
+def do_prediction(image_path, model, use_cpu=False):
     image = Image.open(image_path)
-    image = torchvision.transforms.functional.resize(image, (800, 600))
     image_tensor = torchvision.transforms.functional.to_tensor(image)
 
-    if not use_cpu:
-        # Send to single TPU device
-        device = xm.xla_device()
-        torch_xla._XLAC._xla_set_default_device(str(device))
-        model = model.to(device)
-        image_tensor = image_tensor.to(device)
+    with _bench('inference'):
+        output = model([image_tensor])
+        torch_xla._XLAC._xla_sync_multi(list(output[0].values()), devices=[])
```

Comment:
This may be a better play to try out if `xm.mark_step` works instead of that C binding as Taylan suggested.